dht_bootstrap.go

```diff
@@ -35,6 +37,60 @@ func init() {
 	}
 }
 
+// startSelfLookup starts a go-routine that listens for requests to trigger a self walk on a dedicated channel
+// and then sends the error status back on the error channel sent along with the request.
+// if multiple callers "simultaneously" ask for a self walk, it performs ONLY one self walk and sends the same error status to all of them.
+func (dht *IpfsDHT) startSelfLookup() error {
```

Comment:
Revisiting this, I'm not sure if we need something quite this complicated. We can probably get away with just calling `selfWalk` from within both the refresh function and the "address changed" loop.

(although we'll need to do it asynchronously inside the "address changed" loop)

Note: this code looks fine. I'm just not sure if the complexity is warranted.